core/src/Kokkos_CopyViews.hpp

```diff
@@ -1636,19 +1636,19 @@ inline void deep_copy(
           "match: ");
       message += dst.label();
       message += "(";
-      for (int r = 0; r < dst_type::rank - 1; r++) {
-        message += std::to_string(dst.extent(r));
+      message += std::to_string(dst.extent(0));
+      for (size_t r = 1; r < dst_type::rank; r++) {
```

Comment:
These changes are to avoid warning pointless comparison of unsigned integer with zero when the loop is from 0 to rank-1